### PR TITLE
fix(ngx-jodit-pro): value not changing on reactive form

### PR DESCRIPTION
**Pull request by Duke-of-Karamel**
Changing value via formPatch would not be displayed in editor.

This fix mimics one made almost a year ago to the regular (non-pro) version of this package as I found out during fixing.

### CODE REVIEW

**Comment by Duke-of-Karamel** [comment]
Could you please confirm, that this doesn't happen on your main branch?
I was having this issue even before my changes so I payed no attention to it in the PR.
I have tested it again without them and it seems to still be the case.

**Comment by Duke-of-Karamel** [comment]
No need to test, found it, same bug as above at another place. 
Could you please bump this into npm after merge?

**Comment by julianpoemp** [comment]
Unfortunately setting values doesn't work with your changes. Please run the demo app using "npm run start:pro". Now you see both of the editors without a value set. The left editor should have value "Some text" and the right one "Some text in a reactive form." Yes, it works with the main branch.

Bildschirmfoto 2024-11-20 um 09 34 44

Did you test the change I proposed in my last comment?

**Comment by Duke-of-Karamel** [comment]
I am sorry. I definitely tried your solution at first and without success. But now it seems to work. Although there are still some errors in console.
I am bemused by the fact that jodit-pro requires different treatment from its free variant. The free variant of your library uses setValue in both places I suggested here, but here it seems combination of both value and element html is needed.

While writing this I tried it again and the errors disappeared... Moreover, after removing your suggestion and returning back to my original PR it worked again flawlessly. I suspect there to be some sort of race condition otherwise I cannot explain...

If you look closely, the "null" text still exists in the editor for a split second. (even after implementing your suggestion)

**Comment by julianpoemp** [comment]
Yeah, the jodit-pro package is very strange. Support by Jodit seems dead to me, I already tried to contact the devs several times without response. One important problem is e.g. jodit-pro using ESM (see other issue). The documentation is bad, too.

Back to your issue:
I get errors, too. Something about setting value of undefined. It comes from jodit-pro and it seems that the element, Jodit tries to set the value to, is undefined.

Can you please try to wrap the `this.jodit.setEditorValue(text);`call with try,catch like that:

```typescript
      .subscribe(([[_, initialized], text]) => {
        if (this.jodit && initialized) {
          this.joditContainer.nativeElement.innerHTML = text;
          try {
            this.jodit.setEditorValue(text);
          } catch (e) {
            // ignore
          }
        }
      });
```

It's not the most beautiful solution, but for me that works fine, but I don't know if it works for you using reactive forms.

Another way could be to find out if there is any way to find out if the element for the value is defined or not. There is a property (`this.jodit.isReady`) but I don't know when this property is set to true.

**Comment by Duke-of-Karamel** [comment]
I would appreciate another opinion on this iteration of the solution.
This one awaits the ready promise from jodit. It seems that with this I can come back to my changes without the element HTML.
The .then() is in setter to simply improve readability and show that the function is asynchronous.

Instead of "null" showing for a split second the placeholder now appears before the correct values are set.

**Review by julianpoemp — APPROVED** [review]
It works for me 👍

**Comment by julianpoemp** [comment]
@Duke-of-Karamel I just released ngx-jodit-pro@3.0.1. Thank you for your contribution! 🎉

**Pull request merged**
